docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,382 @@
+
+- The `identityUrl` and `actualUrl` passed to `getAgent()`.
+- The `appId`, `instanceId`, and `instanceUuid` assigned by the DA.
+- An `agentUrl` field with the URL provided in any [`WCP2LoadUrl`](https://fdc3.finos.org/schemas/next/api/WCP2LoadUrl.schema.json) message that was received
+  - Used to skip sub-steps 1-3 in the discovery process described in Step 1.2.
+  - If no [`WCP2LoadUrl`](https://fdc3.finos.org/schemas/next/api/WCP2LoadUrl.schema.json) message was received, omit this field.
+- An `agentType` field which indicates what type of connection to the DA was established
+  - Used to limit the discovery process in Step 1.2 above to only allow agents of the same type.
+  - the `getAgent()` implementation should determine what value to set, from the `WebDesktopAgentType` enumeration for this field based on what happened during the discovery process.
+
+See the [reference documentation for getAgent()](../ref/GetAgent#persisted-connection-data) for further details of the `DesktopAgentDetails` record that should be saved to SessionStorage.
+
+:::tip
+
+SessionStorage is ephemeral, only existing for the duration of the window's life. Hence, there is no concern with the key being overwritten or conflicting with other DAs.
+
+:::
+
+### Step 4: Resolve promise with DesktopAgent interface (step 4)
+
+Resolve the `getAgent()` promise with an object containing either a `DesktopAgent` implementation (that was found at `window.fdc3` or returned by a `failover` function) or a 'Desktop Agent Proxy' implementation (a class implementing the `DesktopAgent` interface that uses the [Desktop Agent Communication Protocol (DACP)](./desktopAgentCommunicationProtocol) to communicate with a Desktop Agent over the `MessagePort`).
+
+### Step 5: Disconnection
+
+Desktop Agent Preload interfaces, as used in container-based Desktop Agent implementations, are usually able to track the lifecycle and current URL of windows that host web apps in their scope. Hence, this is currently no requirement nor means for an app to indicate that it is closing, rather it is the responsibility of the Desktop Agent to update its internal state when an app closes or changes identity.
+
+However, Browser Resident Desktop Agents working with a Desktop Agent Proxy interface may have more trouble tracking child windows and frames. Hence, a specific WCP message ([WCP6Goodbye](https://fdc3.finos.org/schemas/next/api/WCP6Goodbye.schema.json)) is provided for the `getAgent()` implementation to indicate that an app is disconnecting from the Desktop Agent and will not communicate further unless and until it reconnects via the WCP. The `getAgent()` implementation MUST listen for the `pagehide` event from the the HTML Standard's [Page Life Cycle API](https://wicg.github.io/page-lifecycle/spec.html) and send [WCP6Goodbye](https://fdc3.finos.org/schemas/next/api/WCP6Goodbye.schema.json) if it receives an event where the `persisted` property is `false`.
```

Comment:
Do we care that this points to an unofficial, draft specification?  That seems like it might be a big deal to me.

Reply:
This might be a better reference: https://developer.mozilla.org/en-US/docs/Web/API/Window/pagehide_event

Reply:
Not sure how I linked to an unofficial draft, good catch. 

MDN references are useful, but are not the standard either, switched to: https://html.spec.whatwg.org/multipage/nav-history-apis.html#pagetransitionevent and added the MDN reference (as its more useful) and Hrome for developers page on it (way better explanation).

Fixed: 169e6c9fcfdddcd2ca23121f704ded0b182b1aab